dev-python/sphinxcontrib-programoutput/sphinxcontrib-programoutput-0.15.ebuild

```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="Extension to sphinx to include program output"
+HOMEPAGE="https://sphinxcontrib-programoutput.readthedocs.io/en/latest/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+IUSE="doc test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	dev-python/sphinx[${PYTHON_USEDEP}]
+	dev-python/namespace-sphinxcontrib[${PYTHON_USEDEP}]
+"
+DEPEND="
+	${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+	)"
+
+python_compile_all() {
+	if use doc; then
+		python_setup
+		esetup.py build_sphinx
+		HTML_DOCS=( "${BUILD_DIR}/sphinx/html/." )
+	fi
+}
+
+python_test() {
+	esetup.py pytest || die
```

Comment:
`esetup.py` is an internal helper function that dies on its own, no need to prepend it here. You could try running them through `distutils_enable_tests`.